### ISSUE TITLE
Memray hangs when program attempts to import `pyarrow.fs`

### ISSUE THREAD

**Issue by SChakravorti21** [bloomberg/memray#382]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Given the following script (hereafter `script.py`):

```python
from pyarrow import fs
print("hi")
```

Running `memray run script.py` hangs indefinitely and cannot be stopped by sending it `SIGTERM` (I had to suspend it and run `kill -SIGKILL <pid>`).

### Expected Behavior

Expected `memray` to run the script to completion and dump profiling results to a file.

### Steps To Reproduce

1. Create a virtualenv: `python3.10 -m virtualenv venv`
2. Activate and install dependencies: `source venv/bin/activate && pip install pyarrow memray`
3. Run the script _without_ `memray` to confirm that it is valid: `python script.py`
4. Run the script _with_ `memray` to see that it hangs: `memray run script.py`

---

Output of `pip freeze`:

```
Jinja2==3.1.2
markdown-it-py==2.2.0
MarkupSafe==2.1.2
mdurl==0.1.2
memray==1.7.0
numpy==1.24.3
pyarrow==12.0.0
Pygments==2.15.1
rich==13.3.5
```

### Memray Version

1.7.0

### Python Version

3.10

### Operating System

Linux

### Anything else?

I don't know if this helps, but it's worth noting that PyArrow implements a lot of things in native code (C++).

**Comment by SChakravorti21** [comment]
I can confirm that installing from the wheels from the latest CI job resolves this issue for me.

**Comment by godlygeek** [comment]
We've just published wheels for Memray 1.8.0 which contain this fix.
